src/node/index.ts

```diff
@@ -109,6 +116,16 @@ class BeaconNode {
       })
     });
 
+    if (this.conf.validator) {
```

Comment:
I think the validator should be initialized outside node/index.ts. This file should be specific to starting a nodejs-specific beacon node.

We should expose the apis so they can be used externally to initialize the validator.

eg, in some cli file:
```
this.node = new BeaconNode(optionsMap, {logger});
await this.node.start();
if (shouldRunValidator) {
  this.validator = initValidator({apis: this.node.apis, logger});
  await this.validator.start();
}

Reply:
This is for bundled validator so i guess it makes sense to be started here after everything else is initialized

Reply:
Might be better for validator to expose some ValidatorService with start method instead of this initValidator method

Reply:
`initValidator` just initializes the arguments for the constructor of the `validator` class. Validator itself is a service as it has both start and stop method.

Reply:
Yeah this is related to my above question.. could we not in this file just call the beacon chain command and call the validator command? That way we don't need to duplicate logic?

Reply:
I don't think child process can use rpc as instance, we would have to implement some other rpc transport

Reply:
Why not? We could do something sketchy and like re-call the cli with the according flags no?

Reply:
different process == different memory. Maybe a multithreaded version could work with some extra work.
I think we should just stick with the single-process, single-threaded version for now. Keep it minimal, and then get fancy with it later.

Reply:
I could be wrong. I though about using nodejs child processes but afaik we can do it later on, we just need some clean way to start validator haha

Reply:
Yeah im fine with keeping it simple. but ultimately we dont care about sharing memory. Just a cli to start two different processes entirely imo

Reply:
Should I make it resolved for now?

Reply:
@ajoydas 
See my initial comment in this thread,
We want to do the `initValidator` on the cli level, eg: in cli/commands/beacon.ts.
This way we can keep src/node/index.ts only concerned about the beacon chain.